config/db.js

```diff
@@ -2,15 +2,31 @@ const mongoose = require('mongoose')
 mongoose.Promise = require('bluebird')
 const config = require('.')
 
-const connect = () => {
-  mongoose.connect(config.db)
+/**
+ * Connect to mongo DB.
+ * @returns {Promise<*>} - resolves to mongoose instance if successful, rejects otherwise.
+ */
+const connect = async () => {
+  if (mongoose.connection.readyState === 1) {
+    return mongoose
+  }
+  if (mongoose.connection.readyState === 2) {
+    console.log('DB is already connecting, please wait.')
+    return mongoose
+  }
+  let ret
+  try {
+    ret = await mongoose.connect(config.db)
+  } catch (error) {
+    console.error('db connection error:', error.message)
+    throw error
+  }
 
-  const db = mongoose.connection
-  db.on('error', console.error.bind(console, 'connection error:'))
-  db.once('open', () => {
-    console.log('db connection open')
-    // We're connected!
-  })
+  return ret
 }
 
-module.exports = {connect}
+const disconnect = async () => {
+  await mongoose.disconnect()
+}
+
```

Comment:
You're unnecessary forcing async/await a lot 
```js
const disconnect = mongoose.disconnect

Reply:
Resolved